src/axis.ts

```diff
@@ -181,8 +181,8 @@ export const CONDITIONAL_AXIS_PROP_INDEX: Record<
   title: null // title supports signal, let's use it.
 };
 
-export type ConditionalAxisProperty<V extends Value | number[]> = ValueDef<V> & {
-  condition: ConditionalPredicate<ValueDef<V>> | ConditionalPredicate<ValueDef<V>>[];
+export type ConditionalAxisProperty<V extends Value | number[]> = (ValueDef<V> | SignalRef) & {
```

Comment:
This would add signals to every property even if the property does not support signals, though.

Reply:
Or do we always know that the property supports signals because that's how we implement conditionals in the first place?

Reply:
Yes, conditional axis property must either match to an encoding channel (which always support signal) or to a property that supports signal.